src/OpenTelemetry.AutoInstrumentation/Loading/Initializers/GraphQLInitializer.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="GraphQLInitializer.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Reflection;
+using OpenTelemetry.AutoInstrumentation.Configurations;
+using OpenTelemetry.AutoInstrumentation.Plugins;
+
+namespace OpenTelemetry.AutoInstrumentation.Loading.Initializers;
+
+internal class GraphQLInitializer : InstrumentationInitializer
+{
+    private readonly PluginManager _pluginManager;
+    private readonly TracerSettings _tracerSettings;
+
+    public GraphQLInitializer(PluginManager pluginManager, TracerSettings tracerSettings)
+    : base("GraphQL")
+    {
+        _pluginManager = pluginManager;
+        _tracerSettings = tracerSettings;
+    }
+
+    public override void Initialize(ILifespanManager lifespanManager)
+    {
+        var initializerType = Type.GetType("OpenTelemetry.AutoInstrumentation.Initializer, GraphQL");
```

Comment:
I'm a bit disappointed that this code is specific to GraphQL.  I would have expected this to be generic to any library -- just add an internal class with the appropriate name to the library to support auto instrumentation (when not automatically emitted by the library).  In other words, it seems that if I write another library and add a similar class named OpenTelemetry.AutoInstrumentation.Initializer, it's not going to work unless there's specific support added to OpenTelemetry.

Reply:
Unfortunately we cannot scan every loaded library in the startup time, that's going to add up. 

The automated way is to use `OTEL_DOTNET_AUTO_TRACES_ADDITIONAL_SOURCES` https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/config.md#additional-settings. It's expected that every library in the future will use activities by default without custom bootstrapping.

Initializer is already one of many ways to bootstrap activity broadcasting. Definitely the newest way, still not sure how many are going to adapt this pattern.